### PR TITLE
feat: Introduce trait for OpTransaction 

### LINKED ISSUE #11736: Introduce trait for OpTransaction
we have extra helpers for deposit functions:

https://github.com/paradigmxyz/reth/blob/0b6397217ca47fe7cd7f5d30963564f8e63dead8/crates/primitives/src/transaction/mod.rs#L485-L519

these should be converted into a trait on https://github.com/alloy-rs/op-alloy and implemented on 

https://github.com/alloy-rs/op-alloy/blob/56f636e8f624164fdea9c5356ee59b40af8f5e9d/crates/consensus/src/transaction/deposit.rs#L15-L15

### PR DESCRIPTION
**Pull request by hoank101**
Closed #11736

### CODE REVIEW

File: crates/primitives/src/transaction/mod.rs [hunk 1]
```diff
@@ -139,6 +139,34 @@ pub enum Transaction {
     Deposit(TxDeposit),
 }
 
+#[cfg(feature = "optimism")]
+impl DepositTransaction for Transaction {
```
Comment (mattsse) [inline crates/primitives/src/transaction/mod.rs:143]: please use full paths here and avoid the additonal feature gated import

**Review by mattsse — CHANGES_REQUESTED** [review]

File: Cargo.toml
[the remark this answers is not in the document]
Comment (hoank101) [reply Cargo.toml:1]: yeah, just for testing, will update new version

File: crates/optimism/evm/src/execute.rs [hunk 2]
```diff
@@ -4,6 +4,8 @@ use crate::{
     l1::ensure_create2_deployer, OpChainSpec, OptimismBlockExecutionError, OptimismEvmConfig,
 };
 use alloy_primitives::{BlockNumber, U256};
+#[cfg(feature = "optimism")]
+use op_alloy_consensus::DepositTransaction;
```
[the remark this answers is not in the document]
Comment (hoank101) [reply crates/optimism/evm/src/execute.rs]: thank you, i just update code

**Comment by hoank101** [comment]
hi @mattsse can you help me review, now not blocked

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**